### PR TITLE
Np 94/adding drivers configuration

### PR DESCRIPTION
**Pull request by olhabn**
@kgrgreer @jlhughes 
Please take a look and let me know if that's what you had in mind

### CODE REVIEW

File: src/foam/nanos/dig/DIG.js [hunk 1]
```diff
@@ -173,6 +177,26 @@ foam.CLASS({
...
+    },
+    {
+      name: 'exportConfigArray',
+      value: [],
+      hidden: true
+    },
+    {
+      name: 'exportConfigAddOns',
+      factory: function() {
+        this.__context__.extraConfigAddOnDAO.where(this.ContainsIC.create({ arg1: this.AddOn.CONFIG_FOR_CLASS, arg2: this.cls_.id})).select().then((v) => {
```
Comment (kgrgreer) [inline src/foam/nanos/dig/DIG.js]: Better to add:
implements: [ 'foam.mlang.Expressions' ],
And then just do this.CONTAINS_IC( this.AddOn.CONFIG_FOR_CLASS, this.cls_.id )

File: src/foam/nanos/extraconfig/AddOn.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.extraconfig',
+  name: 'AddOn',
+  properties: [
```
Comment (kgrgreer) [inline src/foam/nanos/extraconfig/AddOn.js:11]: add documentation:

File: src/foam/nanos/extraconfig/Config.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.extraconfig',
+  name: 'Config',
+  properties: [
```
Comment (kgrgreer) [inline src/foam/nanos/extraconfig/Config.js:12]: add documentation:

Could you have a more informative name than just 'Config'?

File: src/foam/nanos/extraconfig/Config.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      name: 'exportMetadata',
+      class: 'FObjectProperty',
+      of: 'foam.nanos.extraconfig.AddOn'
+    },
+    {
+      name: 'configValue',
+      factory: function() {     
+          if ( this.exportMetadata.typeOfConfig === 'Boolean' )
+            return false;
+          else if ( this.exportMetadata.typeOfConfig === 'Integer' || this.exportMetadata.typeOfConfig === 'Float' )
```
Comment (kgrgreer) [inline src/foam/nanos/extraconfig/Config.js]: No need for the 'else if' if you've already returned above. Just an 'if' is sufficient.

File: src/foam/nanos/dig/DIG.js (hunk 1 above)
Comment (olhabn) [reply src/foam/nanos/dig/DIG.js]: fixed

File: src/foam/nanos/extraconfig/AddOn.js (hunk 2 above)
Comment (olhabn) [reply src/foam/nanos/extraconfig/AddOn.js:11]: fixed

File: src/foam/nanos/extraconfig/Config.js (hunk 3 above)
Comment (olhabn) [reply src/foam/nanos/extraconfig/Config.js:12]: fixed

File: src/foam/nanos/extraconfig/Config.js (hunk 4 above)
Comment (olhabn) [reply src/foam/nanos/extraconfig/Config.js]: fixed

**Review by jlhughes — APPROVED** [review]

**Pull request merged**